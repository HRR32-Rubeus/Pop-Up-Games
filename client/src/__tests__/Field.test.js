import Field from '../components/field/Field.jsx';
import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Field component', () => {
  it('Field: renders correctly', () => {
    const props = { target: { field: 1 }, changeTarget: jest.fn() };
    const tree = renderer.create(<Field {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

//issues with testing axios functions because they are dependent on state

// it ('returns field data when getFieldData is called', done => {
// 	const wrapper = shallow(<Field/>);
// 	const instance = wrapper.instance()
// 	const mock = new MockAdapter(axios);
// 	const mockData = {field: {id:1, fieldName: 'Grass Field1'}}
// 	mock.onGet('/api/field', {params: {id: 1}}).reply(200,mockData)

// 	var contextObj = {state: {fieldId:1}}
// 	Field.prototype.getFieldData().then(response => {
// 		expect(response.data).toEqual(mockData);
// 		done();
// 	});

// })
