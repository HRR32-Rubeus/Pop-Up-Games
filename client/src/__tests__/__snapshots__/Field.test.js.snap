// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field component Field: renders correctly 1`] = `
<div>
  Loading
</div>
`;
